src/main/java/com/gard/testbed/activitySystem/State.java

```diff
@@ -16,21 +16,32 @@ public State(String stateId) {
     }
 
     @Override
-    public String getId() {
-        return id;
+    public boolean isCompleted(IBlackboard player) {
+        return transitions.stream().allMatch(e -> e.isComplete(player));
+    }
+
+    @Override
+    public boolean isActive() {
+        return false;
     }
```

Comment:
Agreed - that was the assumption I was working under. I am missing something in understanding how this structure will give us the functionality we are ultimately after then - to find out where the player is in the progression map (ie which state he is in, and therefore the options available (outgoing transitions)) should we traverse the whole map (following live transitions) every time it is updated / the info is requested?